### ISSUE TITLE
Bad UX for Checkbox Button on mobile browser

### ISSUE THREAD

**Issue by Juchar** [ng-bootstrap/ng-bootstrap#2484]
### Bug description:
Using the checkbox buttons on a mobile browser (Chrome Browser 67.0.3396.87 on Android) if you uncheck/check a button it will not show the proper styling as it seems to be still hovered. 

This is resulting in really bad UX due to the fact that for the overall user the button still seems to be unchecked.

You can see in the below screenshot the most right button is checked, but still showing the unchecked styling:

![screenshot](https://user-images.githubusercontent.com/14292982/42077516-7c157998-7b79-11e8-935c-ba892d869f88.png)

### Link to minimally-working plunker that reproduces the issue:

You can check this by simply accessing the demo here using a mobile browser:
[https://ng-bootstrap.github.io/#/components/accordion/examples](https://ng-bootstrap.github.io/#/components/accordion/examples)

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 6.0.6
ng-bootstrap: 2.1.2
Bootstrap: 4.1.0

**Comment by Juchar** [comment]
By the way: from my point of view this is also confusing people on the desktop browser, especially when using the `btn-outline` style I assume the people would expect the button to be instantly not filled after clicking it and not after having to move the mouse away first.

**Comment by pkozlowski-opensource** [comment]
> This is resulting in really bad UX due to the fact that for the overall user the button still seems to be unchecked.

Totally agree! Unfortunately this is what we've got from Bootstrap CSS, you can see the same behaviour  in the Bootstrap demo site: http://getbootstrap.com/docs/4.1/components/buttons/#checkbox-and-radio-buttons

This steams from the fact that Bootstrap adds darker background for buttons that are focused. I agree with you that there should be more pronounced difference between checked / unchecked and hoovered.

Would you mind opening an issue in https://github.com/twbs/bootstrap/issues ?

**Comment by Juchar** [comment]
@pkozlowski-opensource Done, I created the issue [#26804](https://github.com/twbs/bootstrap/issues/26804).

**Comment by Juchar** [comment]
I have implemented a workaround that might be useful for others struggling with this.

## Directive
```Typescript
@Directive({
  selector: "[ngbCheckboxButtonLabel]"
})
export class NgbCheckBoxButtonLabelDirective {

  private nativeElement: any;

  constructor(private renderer: Renderer2, private  elementRef: ElementRef) {
    this.nativeElement = this.elementRef.nativeElement;
  }

  @HostListener("click")
  onClick() {
    const nativeElement = this.elementRef.nativeElement;
    this.renderer.addClass(nativeElement, "clicked");
  }

  @HostListener("mouseleave")
  onMouseLeave() {
    const nativeElement = this.elementRef.nativeElement;
    this.renderer.removeClass(nativeElement, "clicked");
  }
}
```

## SCSS

```scss
/* Custom adaptations for existing elements in (ng-)bootstrap. */
@import '../../node_modules/bootstrap/scss/functions';

@each $color, $value in $theme-colors {
  .btn-outline-#{$color}.clicked:not(.active) {
    color: $value;
    background: transparent none;
  }

  .btn-#{$color}.clicked:not(.active) {
    color: color-yiq($value);
    @include gradient-bg($value);
    border-color: $value;
  }
}
```

## Usage
```html
<label class="btn btn-outline-primary" ngbButtonLabel ngbCheckboxButtonLabel>
    <input type="checkbox" ngbButton>Button
</label>
```

## What happens
The `clicked` style gets applied to the button after a click as long as the mouse does not leave it and the according styling is applied. As soon as the user once left the button, the `clicked` style gets removed and the hover style is applied again to the button on hovering.

**Comment by maxokorokov** [comment]
Buttons were removed in v13 → https://ng-bootstrap.github.io/releases/12.x/#/components/buttons/overview
